packages/vscode-apollo/src/debug.ts

```diff
@@ -0,0 +1,74 @@
+import { window, workspace, OutputChannel } from "vscode";
+
+/**
+ * for errors (and other logs in debug mode) we want to print
+ * a stack trace showing where they were thrown. This uses an
+ * Error's stack trace, removes the three frames regarding
+ * this file (since they're useless) and returns the rest of the trace.
+ */
+const createAndTrimStackTrace = () => {
+  let stack: string | undefined = new Error().stack;
+  // remove the lines in the stack from _this_ function and the caller (in this file) and shorten the trace
+  return stack && stack.split("\n").length > 2
+    ? stack
+        .split("\n")
+        .slice(3, 7)
+        .join("\n")
+    : stack;
+};
+
+export class Debug {
```

Comment:
Can we just require this class be instantiated with an `OutputChannel` so we don't have to check the existence of `this.outputConsole` everywhere?

Reply:
Since we're never explicitly initializing the class, just using the static functions on it, I don't think so. I'll refactor to exit early instead